modules/bedtools_intersectNeg.cfmod.pl

```diff
@@ -0,0 +1,118 @@
+# it under the terms of the GNU General Public License as published by   #
+# the Free Software Foundation, either version 3 of the License, or      #
+# (at your option) any later version.                                    #
+#                                                                        #
+# Cluster Flow is distributed in the hope that it will be useful,        #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of         #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          #
+# GNU General Public License for more details.                           #
+#                                                                        #
+# You should have received a copy of the GNU General Public License      #
+# along with Cluster Flow.  If not, see <http://www.gnu.org/licenses/>.  #
+##########################################################################
+
+my %requirements = (
+	'cores' 	=> ['2', '4'],
+	'memory' 	=> ['8G', '16G'],
+	'modules' 	=> 'BEDTools',
+	'time' 		=> sub {
+
+		my $cf = $_[0];
+		my $num_files = $cf->{'num_starting_merged_aligned_files'};
+		# Default to 1 if none were parsed for whatever reason
+		$num_files = ($num_files > 0) ? $num_files : 1;
+		# To be safe, we set the time limit to 1 hour per bam file.
+		return CF::Helpers::minutes_to_timestamp ($num_files * 1 * 60);
+	}
+);
+
+# The help text
+my $helptext = "".("-"x15)."\n Bedtools intersectNeg\n".("-"x15)."\n
+Takes a bam file and a bed file with regions, given by the flag blacklistFile. 
+Filters the bam file, so the only reads not present in the blacklist regions 
+are included in the output. Usees bedtools intesect for this.\n\n";
+
+# Start your engines...
+my %cf = CF::Helpers::module_start(\%requirements, $helptext);
+
+# Print version information about the program to be executed.
+warn "---------- Bedtools version information ----------\n";
+warn `bedtools --version`;
+warn "\n------- End of bedtools version information ------\n";
+
+# Open up our run file in append mode
+open (RUN,'>>',$cf{'run_fn'}) or die "###CF Error: Can't write to $cf{run_fn}: $!";
+
+# Go through each file and run our command
+foreach my $file (@{$cf{'prev_job_files'}}){
+  # Start the clock...
+  my $timestart = time;
+
+  # Check that input is bam file
+  if($file !~ /\.bam$/i){
+      warn "\n###CF Error! bedtools_intersectNeg failed: bam file expected, got $file\n\n";
+      next;
+  }
+
+  # Generate a nice output file name
+  (my $output_fn = $file) =~ s/\.bam//i;
+  $output_fn .= "_noblacklist.bam";
+
+
+  # Check if a file with blacklist regions was provided.
+  my $blacklistFile;
+  if(defined($cf{'params'}{'blacklistFile'})){
+      $blacklistFile = $cf{'params'}{'blacklistFile'};
+      warn "###CF bedtools intersectNeg: Using blacklist file $blacklistFile.\n";
+  } else {
+      warn "###CF bedtools intersectNeg: No blacklist file given. Skipping this step.\n";
```

Comment:
Nice idea to not kill the entire pipeline!

If you put the word `error` somewhere in this string, the summary script [will recognise it](https://github.com/ewels/clusterflow/blob/master/modules/cf_runs_all_finished.cfmod.pl#L139-L144) and the summary e-mail will give a prominent warning / error message about it. Nothing fancy needed, eg:

```perl
warn "###CF bedtools intersectNeg: Error - no blacklist file given. Skipping this step.\n";
```